src/utils/renderChunks.ts

```diff
@@ -111,6 +111,10 @@ async function transformChunk(
 ) {
 	let map: SourceMap | null = null;
 	const sourcemapChain: DecodedSourceMapOrMissing[] = [];
+	const entryModule = usedModules.find(module => module.info.isEntry);
+	if (entryModule && entryModule.shebang !== undefined && ['cjs', 'es'].includes(options.format)) {
+		magicString.prepend(`#!${entryModule.shebang}\n`);
+	}
```

Comment:
Why do it here and not in `Chunk.render` where everything else is rendered?
More specifically, it feels to me like something I would search for in the "finalizers", i.e. `es.ts` and `cis.ts`. You could add the shebang as another property to the `FinaliserOptions` and only render it for those two formats.
Another point to consider: Especially when using manual chunks, a chunk might have several entry points. Those do not matter. The only module that matters is the `facadeModule`, i.e. the module this chunk pretends to be. See also the `isEntryFacade` property.

Reply:
Ah, I see, `Chunk.render` is a more appropriate place.

Reply:
> You could add the shebang as another property to the `FinaliserOptions` and only render it for those two formats.

For rendering `shebang` before `banner`, I prepend `shebang` after `banner` in `Chunk.render`.

Reply:
Ah you are right, I missed that banner is attached outside the finalizers. Good work then!